File: 30 Days of Code/C#/15 - Day 14 - Scope.cs
// ========================
//       Information
// ========================

// Direct Link: https://www.hackerrank.com/challenges/30-scope/problem
// Difficulty: Easy
// Max Score: 30
// Language: C#

// ========================
//         Solution
// ========================

using System;
using System.Linq;

class Difference {
	private int[] elements;
	public int maximumDifference;

	// Add your code here
	public Difference(int[] elements) {
		this.elements = elements;
	}

	public int computeDifference() {
		int maxElement = elements[0];
		int minElement = elements[0];

		for (int i = 0; i < elements.Length; i++) {
			if (elements[i] < minElement) {
				minElement = elements[i];

			}
			if (elements[i] > maxElement) {
				maxElement = elements[i];
			}
		}
		maximumDifference = Math.Abs(maxElement - minElement);
		return maximumDifference;
	}
}

class Solution {
	static void Main(string[] args) {
		Convert.ToInt32(Console.ReadLine());

		int[] a = Console.ReadLine().Split(' ').Select(x = >Convert.ToInt32(x)).ToArray();

		Difference d = new Difference(a);

		d.computeDifference();

		Console.Write(d.maximumDifference);
	}
}

File: 30 Days of Code/Python/15 - Day 14 - Scope.py
# ========================
#       Information
# ========================

# Direct Link: https://www.hackerrank.com/challenges/30-scope/problem
# Difficulty: Easy
# Max Score: 30
# Language: Python

# ========================
#         Solution
# ========================

class Difference:
    def __init__(self, a):
        self.__elements = a
        self.maximumDifference = 0

    # Add your code here
    def computeDifference(self):
        l = len(a)
        for i in range(0, l):
            for j in range(i + 1, l):
                difference = abs(a[i] - a[j])
                self.maximumDifference = max(difference, self.maximumDifference)

# End of Difference class

_ = input()
a = [int(e) for e in input().split(' ')]

d = Difference(a)
d.computeDifference()

print(d.maximumDifference)
